### PR TITLE
Tweaked data loading for Agents Show

### PR DESCRIPTION
**Pull request by keithpitt**
Tweaked loading of the Agents Show page so have the browser just hang on you while it's performing necessity queries to load the page.

### CODE REVIEW

**Comment by ticky** [comment]
I think my main question with this is, what happens if Shopify load this page? Are we then requesting _all_ the agent data repeatedly?

**Comment by ticky** [comment]
wait, never mind, I made assumptions about your approach ;P

File: app/components/agent/Show.js [hunk 1]
```diff
@@ -34,11 +35,22 @@ class AgentShow extends React.Component {
   };
 
   componentDidMount() {
-    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
+    // Only bother setting up the delayed load and refresher if we've got an
+    // actual agent to play with.
+    if(this.props.agent && this.props.agent.id) {
+      this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
```
Comment (ticky) [inline app/components/agent/Show.js]: wanna update this to do `setTimeouts` instead as part of this?

File: app/components/agent/Show.js [hunk 2]
```diff
@@ -34,11 +35,22 @@ class AgentShow extends React.Component {
...
+      // to show this page.
+      this.props.relay.setVariables({ isMounted: true });
+    }
   }
 
   componentWillUnmount() {
-    clearInterval(this._agentRefreshInterval);
+    if(this._agentRefreshInterval) {
+      clearInterval(this._agentRefreshInterval);
+    }
```
Comment (ticky) [inline app/components/agent/Show.js]: `clearInterval` doesn’t actually care if it’s given nothing! which is weird, I guess

File: app/components/agent/Show.js (hunk 1 above)
Comment (keithpitt) [inline app/components/agent/Show.js]: I was gonna, but then `this.props.relay.setVariables({ isMounted: true })` needs to be in the `if` as well, so I figured this'd be the best place.

File: app/components/agent/Show.js (hunk 2 above)
Comment (keithpitt) [inline app/components/agent/Show.js]: Oh lol! Will remove the check.

**Pull request merged**